extensions/functions_string.yaml

```diff
@@ -133,6 +133,55 @@ scalar_functions:
           dotall:
             values: [ DOTALL_DISABLED, DOTALL_ENABLED ]
         return: "string"
+  -
+    name: regexp_match_substring_all
+    description: >-
+      Extract all substrings that match the given regular expression pattern. The regular expression
+      pattern should follow the International Components for Unicode implementation
```

Comment:
This standard includes negative lookahead assertions which is known to be quite slow.  To combat this Google wrote https://github.com/google/re2.  It's not quite an option but it definitely is a behavior that a dialect would have so I'm not sure we can code it but we could add something to the description so that users might be aware of the possibility.

Reply:
Why wouldn't regex dialect be an option?

Reply:
International Components for Unicode (ICU) has a regex dialect standard that a some other SQL dialects follow.  The description includes a link to the standard.  I originally went with that in [this PR](https://github.com/substrait-io/substrait/pull/281).  I think MySQL and IBM follow it.  I supposed the regex dialect may not matter so long as the producer is able to generate a substrait plan though?

Reply:
I can open up a separate issue to discuss this.  The other existing regex functions state this as well, and I was just continuing with that.